source/agora/consensus/validation/Block.d

```diff
@@ -112,6 +116,138 @@ public string isInvalidReason (const ref Block block, in ulong prev_height,
     return null;
 }
 
+/*******************************************************************************
+
+    Check the validity of a genesisBlock.
+
+    A genesisBlock is considered valid if:
+        - Block height is 0
+        - The previous block hash of the block must be empty
+        - The number of transactions in the block is above zero and less than
+          Block.TxsInBlock
+        - Transactions are ordered by their hash value
+        - All the transactions pass validation, which implies:
+            - Transaction type in block should only be Payment or Freeze
+            - There is only one input for the transaction
+            - The signature value of the input of the transaction is genuine
+            - There is more than one output of the transaction
+            - The output of the transaction should not be empty at address
+            - The output.value of the transaction must be in accordance
+              with the `Amount` rule
+        - The merkle root in the header matches the re-built merkle tree root
+        - All the enrollments pass validation, which implies:
+            - The Enrolments should be arranged in ascending order by utxo_key
+            - Signature are authentic
+            - The enrollments refer to freeze tx's in this block
+            - The freeze tx's spend an output which must be found with
+              the findUTXO() delegate
+            - Freeze transactions referenced in registration should not be
+              double-spend
```

Comment:
This is repeating much of the `Block` part, right ? I'd rather have this say "Follow the Block rules except for the following:" so it's easier on the reader.